### ISSUE TITLE
"undefined reference" in linking phase

### ISSUE THREAD

**Issue by taranion** [gluonhq/client-maven-plugin#294]
I have a project which compiles just fine, but fails with several "undefined references" in the linking phase. These references refer to symbols in sun_awt_image or com-sun_imageio. Since I don't access those symbols myself, I assume that some JAR file I pull in through dependencies (likely javax.activation) references them

Two questions:

1. Why is the missing reference not detected during compilation?
2. Any ideas how I can work around this?

GraalVM 21.0.0
Client plugin: 0.1.36

[process-link-1612442341833.log](https://github.com/gluonhq/client-maven-plugin/files/5925803/process-link-1612442341833.log)
[process-compile-1612442339132.log](https://github.com/gluonhq/client-maven-plugin/files/5925804/process-compile-1612442339132.log)
[client-debug0.log](https://github.com/gluonhq/client-maven-plugin/files/5925805/client-debug0.log)

**Comment by taranion** [comment]
It looks like these are all private native methods of JPEGImageReader or JPEGImageWriter.

And I correct my assumption about javax.activation - it seems the problematic artifact that causes this is runtime dependency for org.glassfish.jaxb:jaxb-runtime (tried 2.3.1 and 2.4.0)

**Comment by taranion** [comment]
Possibly related: When targeting Android instead of Desktop, compiling and linking passes without problems and I get a resulting APK.
When running this on my phone, it fails with
```
02-22 10:14:44.742 18051 18051 E AndroidRuntime: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "JNI_OnLoad_awt" referenced by "/data/app/~~mrpc6-gXESDmscHfLFnwFA==/de.plusnet.centraflexmobile-uOlwdrk-6KfSrTY0G2L0PQ==/lib/arm64/libsubstrate.so"...
02-22 10:14:44.742 18051 18051 E AndroidRuntime: 	at java.lang.Runtime.loadLibrary0(Runtime.java:1087)
02-22 10:14:44.742 18051 18051 E AndroidRuntime: 	at java.lang.Runtime.loadLibrary0(Runtime.java:1008)
02-22 10:14:44.742 18051 18051 E AndroidRuntime: 	at java.lang.System.loadLibrary(System.java:1664)
02-22 10:14:44.742 18051 18051 E AndroidRuntime: 	at com.gluonhq.helloandroid.MainActivity.surfaceCreated(MainActivity.java:104)
```

**Comment by jperedadnr** [comment]
Should be fixed with plugin 0.1.39
